src/Utils/dialogEditing.ts

```diff
@@ -506,7 +506,7 @@ export async function getPlaceholderAPIAction(
         if (!newAction) {
             throw new Error("Failed to create placeholder API")
         }
-
+        actions.push(newAction)
```

Comment:
I think it would be better to create a local list rather than push these on to the passed list.  That could have unintended side effects

Reply:
Nice to see preference for immutability now. I also agree that it's best to use pure functions and avoid mutations when possible. This is part of philosophy of react, redux, and declarative based systems.

Reply:
Would taking a copy of the actions and entities in the ctor be what you had in mind ?

eg
        this.actions = actions
=>
        this.actions = [...actions]

Reply:
To Matt's point, the only place we know whether or not a given entity or action has been created is in the utility functions that create them.  As currently implemented, we can't tell from the return values whether or not the element being returned is new or memoized.

Reply:
Per conversation, if we're going to keep the mutation since alternatives are messy, then would rename the function. The name `getPlaceholderAPIAction` doesn't imply it also modifies list, perhaps `addPlaceholderAPIAction`

Reply:
@mattmazzola good suggestion, renaming to getOrCreate...

Reply:
`getOrCreate` doesn't imply the `actions.push(newAction)` though.  Sign of methods doing too many things when they are hard to name.  This point is caller should be aware that it modifies list.  

It didn't realize this can also return `null` so it's not even a getter... ugh. Do whatever is easier then cause there is no good solution.

Reply:
Ok - updating comments and merging.

Reply:
@mattmazzola @LarsLiden - you good with these changes ?

Reply:
I suppose a reluctant "yes". I don't like it as name is still misleading and function still mutates it's argument which is likely to cause issues later. Yea, there is a comment but that doesn't make it much better. That being said it doesn't seem like there are any easy solutions because there are many problems surrounding so fixing this wouldn't help much.  This is all part of problem of bad code getting committed. Your task is seemingly small and it would seem unfair to require fixing all this so you go ahead just do the small change. Enough of these add up, and here we are...

Reply:
Alright; we can chat process more tomorrow, I'll go ahead and merge this to unblock Swadheen's testing.  Thanks!